robottelo/cli/factory.py

```diff
@@ -3965,33 +3777,30 @@ def setup_cdn_and_custom_repos_content(
         repos=repos,
         download_policy=download_policy
     )
-    # Create a content view
-    content_view_id = make_content_view({u'organization-id': org_id})['id']
-    # Add repositories to content view
-    for repo_info in repos_info:
-        ContentView.add_repository({
-            u'id': content_view_id,
+    if default_cv:
+        activation_key = make_activation_key({
             u'organization-id': org_id,
-            u'repository-id': repo_info['id'],
+            u'lifecycle-environment': 'Library',
+        })
+        content_view = ContentView.info({
+            u'organization-id': org_id,
+            u'name': u'Default Organization View'
+        })
+    else:
+        # Create a content view
+        content_view = make_content_view({u'organization-id': org_id})
+        # Add repositories to content view
+        for repo_info in repos_info:
+            ContentView.add_repository({
+                u'id': content_view['id'],
+                u'organization-id': org_id,
+                u'repository-id': repo_info['id'],
+            })
+        activation_key = make_activation_key({
+            u'organization-id': org_id,
+            u'lifecycle-environment-id': lce_id,
+            u'content-view-id': content_view['id'],
```

Comment:
@rplevka is this really going to work without publishing and promoting? (why that functionality exists ever), perhaps this need a more accurate testing for custom and RH repos (note we have to use the user 's path to setup entities).

Reply:
yes, it does (see my test results). Every repo you add is automatically in a default organizational view, which is in Library.
The feature fo content views exists to enable further content filtering, etc. 99% of our tests don't need the constant publishes and promotes.

Reply:
@rplevka I can understand  your when the Library env is used, but when we supply lvce other than Library, how that will work without publishing and promoting?

Reply:
@ldjebran, if you supply lce_id without setting the default_cv flag, then custom CV will be created just as in the original commit. Otherwise this is overriden and Library LCE is used:
https://github.com/rplevka/robottelo/blob/9a28aa7426fca864142e29c5681c643380742f41/robottelo/cli/factory.py#L3785

..i can update the docstring to make this more clear, or i can throw a `TypeError` to point that out, same way, as i do if None of these parameters are specified.

Reply:
but this content view here https://github.com/rplevka/robottelo/blob/9a28aa7426fca864142e29c5681c643380742f41/robottelo/cli/factory.py#L3793 is not published nor promoted as the code was removed

Reply:
@ldjebran oh yes, sorry - missed this while moving it away. will bring it back